### ISSUE TITLE
Rspamd writes negative message rate in RRD on `reload`.

### ISSUE THREAD

**Issue by moisseev** [vstakhov/rspamd#894]
**Anti**-spam flow:
![Anti-spam flow displayed on rspamd web interface graph](https://cloud.githubusercontent.com/assets/2275981/17929007/5065b99c-6a06-11e6-840c-86bcf96c9358.png)

~~~xml
        <!-- Round Robin Archives -->
        <rra>
                <cf>AVERAGE</cf>
                <pdp_per_row>60</pdp_per_row> <!-- 60 seconds -->

                <params>
                <xff>5.000000000e-01</xff>
                </params>
                <cdp_prep>
                        <ds>
                        <primary_value>0.000000000e+00</primary_value>
                        <secondary_value>0.000000000e+00</secondary_value>
                        <value>0.000000000e+00</value>
                        <unknown_datapoints>0</unknown_datapoints>
                        </ds>
                        <ds>
                        <primary_value>0.000000000e+00</primary_value>
                        <secondary_value>0.000000000e+00</secondary_value>
                        <value>0.000000000e+00</value>
                        <unknown_datapoints>0</unknown_datapoints>
                        </ds>
                        <ds>
                        <primary_value>0.000000000e+00</primary_value>
                        <secondary_value>0.000000000e+00</secondary_value>
                        <value>0.000000000e+00</value>
                        <unknown_datapoints>0</unknown_datapoints>
                        </ds>
                        <ds>
                        <primary_value>0.000000000e+00</primary_value>
                        <secondary_value>0.000000000e+00</secondary_value>
                        <value>0.000000000e+00</value>
                        <unknown_datapoints>0</unknown_datapoints>
                        </ds>
                </cdp_prep>
                <database>
                       <!-- 2016-08-23 14:17:00 MSK / 1471951020 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>1.647327097e-02</v></row>
 
...

                       <!-- 2016-08-24 13:06:00 MSK / 1472033160 --> <row><v>1.652565325e-02</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>4.969805634e-02</v></row>
                        <!-- 2016-08-24 13:07:00 MSK / 1472033220 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>6.631641918e-02</v></row>
                        <!-- 2016-08-24 13:08:00 MSK / 1472033280 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v></row>
                        <!-- 2016-08-24 13:09:00 MSK / 1472033340 --> <row><v>0.000000000e+00</v><v>1.657767779e-02</v><v>0.000000000e+00</v><v>4.950222485e-02</v></row>
                        <!-- 2016-08-24 13:10:00 MSK / 1472033400 --> <row><v>-5.349789184e-02</v><v>-1.024320837e-02</v><v>0.000000000e+00</v><v>-6.152257562e-01</v></row>
                        <!-- 2016-08-24 13:11:00 MSK / 1472033460 --> <row><v>0.000000000e+00</v><v>-3.978157266e-05</v><v>0.000000000e+00</v><v>1.656243828e-02</v></row>
                        <!-- 2016-08-24 13:12:00 MSK / 1472033520 --> <row><v>0.000000000e+00</v><v>-6.148320889e-05</v><v>0.000000000e+00</v><v>-1.658403760e-02</v></row>
                        <!-- 2016-08-24 13:13:00 MSK / 1472033580 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>1.657067120e-02</v></row>
                        <!-- 2016-08-24 13:14:00 MSK / 1472033640 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>6.623383890e-02</v></row>
                        <!-- 2016-08-24 13:15:00 MSK / 1472033700 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v></row>
                        <!-- 2016-08-24 13:16:00 MSK / 1472033760 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v></row>
                        <!-- 2016-08-24 13:17:00 MSK / 1472033820 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v></row>
                        <!-- 2016-08-24 13:18:00 MSK / 1472033880 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>1.656090802e-02</v></row>
                        <!-- 2016-08-24 13:19:00 MSK / 1472033940 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>1.664813832e-02</v></row>
                        <!-- 2016-08-24 13:20:00 MSK / 1472034000 --> <row><v>0.000000000e+00</v><v>-1.653363847e-02</v><v>0.000000000e+00</v><v>-1.149491137e-01</v></row>
                        <!-- 2016-08-24 13:21:00 MSK / 1472034060 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v></row>
                        <!-- 2016-08-24 13:22:00 MSK / 1472034120 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>1.658143698e-02</v><v>0.000000000e+00</v></row>
                        <!-- 2016-08-24 13:23:00 MSK / 1472034180 --> <row><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v><v>0.000000000e+00</v></row>
~~~

**Comment by moisseev** [comment]
Negative spikes happens on rspamd `reload`.

**Comment by mnik247** [comment]
+1.
I have negative value too - every 24 hours.

**Comment by moisseev** [comment]
@mnik247 Seems like in your case newsyslog/logrotate sends `HUP` to rspamd. You can use `SIGUSR1` instead.

**Comment by moisseev** [comment]
Fixed with 74532cb312603671c51008d20e1f66eeee8ecc0d

To fix broken RRD you can replace negative RRA values with `NaN`:
~~~sh
rrdtool dump /var/db/rspamd/rspamd.rrd > rspamd.rrd.xml.orig && \
sed -E '/<\/row>$/ { s|<v>-[^<]+</v>|<v>NaN</v>|g; }' rspamd.rrd.xml.orig > rspamd.rrd.xml && \
rrdtool restore -f rspamd.rrd.xml /var/db/rspamd/rspamd.rrd
~~~
It can be done without stopping `rspamd` service.
